design/design-doc-resource-configurations.md

```diff
@@ -0,0 +1,838 @@
+  has been provisioned.
+* When a resource class references a statically provisioned managed resource
+  every managed resource that is dynamically provisioned using that class will
+  reference that specific managed resource. For example if a `GKEClusterClass`
+  references a `Subnetwork` then every `GKECluster` dynamically provisioned
+  using said class will attempt to share said `Subnetwork`, despite it often
+  being desirable to create a unique `Subnetwork` for each dynamically
+  provisioned `GKECluster`.
+
+The one-to-one relationship between resource claims and resource classes thus
+weakens portability, separation of concerns, and support for [GitOps]. An
+infrastructure operator can publish a resource class representing a single
+managed resource that an application operator may dynamically provision, but in
+the likely event that managed resource requires supporting managed resources to
+function usefully the application operator must ask an infrastructure operator
+to provision them.
+
+Furthermore, defining a core set of portable resource claims has begun to limit
+Crossplane. Resource claims are subject to the [lowest common denominator]
+problem of multi-cloud; when a claim may provide configuration inputs that may
+be used to match or provision many kinds of managed resource it may support only
+the settings that apply to _all_ compatible managed resources. This is in part
+why Crossplane defines relatively few resource claim kinds. Meanwhile, it's
+possible that an infrastructure operator deploys Crossplane to provide an
+opinionated abstraction for the application operators in their organisation and
+that said organisation only uses AWS. If this organisation values Crossplane's
+separation of concerns but does not need its portability there is no reason that
+its application operators should be limited to resource claims that are
+constrained by supporting all possible providers.
+
+## Goals
+
+The goals of this proposal are to enable infrastructure operators to publish
+resource classes that represent _groups_ of infrastructure, and to define in
+configuration (not code) how those resource classes relate to resource claims.
+
+It is important that the design put forward:
+
+* Support user defined resource claims with arbitrary configuration parameters.
+* Ensures claim-kind-to-class-kind and claim-kind-to-resource-kind compatibility
+  is discoverable through configuration, not code.
+* Ensures that several interdependent resources submitted to Crossplane
+  simultaneously may eventually resolve to functioning infrastructure.
+* Limits unnecessary API churn, especially to `v1beta1` managed resource APIs
+  which must maintain backward compatibility.
+* Maintains the high-fidelity, composable nature of managed resources.
+* Maintain support for both static and dynamic provisioning.
+* Follow [declarative configuration best practices] and Kubernetes API
```

Comment:
Looks like this link did not get assigned a url below.

Reply:
Should be fixed now.